companion-app/src/Components/GroupPlay.js

```diff
@@ -1,12 +1,257 @@
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            usernames: {
+                user0: "",
+                user1: "",
+                user2: "",
+                user3: "",
+                user4: ""
+            },
+            numUsers: 0,
+            userIDs: [],
+            groupFileName: "",
+            play: "false",
+            stop: "false"
+        }
+
+        this.addUsersToGroup = this.addUsersToGroup.bind(this);
+        this.shareFile = this.shareFile.bind(this);
+    }
+
+    addUsersToGroup = async (event) => {
+        event.preventDefault();
+
+        this.setState({ userIDs: [] });
+
+        let usersObj = this.state.usernames;
+        let usersToAdd = Object.keys(usersObj).map(function(key) {
+            return usersObj[key];
+        });
+
+        let apiName = "AdminQueries";
+        let path = "/getUser"
+
+        let userCount = 0;
+        let userAdded = false;
+        for (const i in usersToAdd) {
+            if(usersToAdd[i] !== "") {
+                let myInit = {
+                    queryStringParameters: {
+                        "username" : usersToAdd[i]
+                    },
+                    headers: {
+                        'Content-Type' : 'application/json',
+                        Authorization: `${(await Auth.currentSession()).getAccessToken().getJwtToken()}`
+                    }
+                }
+
+                let userDetails = await API.get(apiName, path, myInit);
+                let userIDTemp = this.state.userIDs.concat();
+                userIDTemp.push(userDetails['UserAttributes'][0]["Value"])
+                this.setState({
+                    userIDs: userIDTemp
+                });
+
+                userAdded = true;
+                ++userCount;
+
+                this.setState({
+                    numUsers: userCount
+                })
+            } 
+        }
+
+        console.log(this.state.userIDs);
+
+        if(!userAdded) {
+            window.alert("Please specify at least one user");
+        }
+    }
+
+    shareFile = async () => {
+        let groupUsers = this.state.userIDs;
+        for (const i in groupUsers) {
+            groupPacket.group[i].userID = groupUsers[i];
+        }
+
+        groupPacket.filename = this.state.groupFileName;
+        
+        // for debugging
+        console.log(groupPacket);
+
+        // here for reference, if this is still exists please remove it
+/* 
+        Storage.put(jsonFilePrefix + '.json', packet)
+        .then (result => console.log(result))
+        .catch(err => console.log(err));
+*/
+        // assign json file prefixes as per each user's UUID
+        for (const i in groupUsers) {
+            let prefix = groupUsers[i];
+
+            Storage.put(prefix + '.json', groupPacket)
+            .then(result => console.log(result))
+            .catch(err => console.log(err));
+        }
+    }
+
+    setAudioState = () => {
```

Comment:
Toggle functionality seems strange in a "set" function

Reply:
changed naming